AppCenter/AppCenter/Internals/Util/MSUtility+Application.m

```diff
@@ -47,6 +47,11 @@ + (UIApplication *)sharedApp {
 
 #if TARGET_OS_OSX
 + (MSApplicationState)sharedAppState {
+
+  // UI API (isHidden) cannot be called from a background thread.
+  if (![NSThread isMainThread]) {
```

Comment:
Can we get the state from UI thread? I know it isn't a good idea because it will be a synchronous call and block the thread.

Reply:
No, main thread can be blocked by waiting this so queues doesn't work here.

See https://github.com/microsoft/appcenter-sdk-apple/pull/1707#issuecomment-516429117 with my variants